src/SystemReport.php

```diff
@@ -228,21 +212,15 @@ public function request( $response, $extra = null ) {
 	 *
 	 * @hook woocommerce_cleanup_logs
 	 */
-	public static function remove_old_entries() {
-		$registry = get_option( self::REGISTRY_OPTION, array() );
-		foreach ( $registry as $id ) {
-			$reports = json_decode( get_option( 'krokedil_support_' . $id, '[]' ), true );
-			if ( empty( $reports ) ) {
-				continue;
-			}
+	public function remove_old_entries() {
+		$retention_period = wc_get_container()->get( Settings::class )->get_retention_period();
```

Comment:
This is calling a internal class and method from WooCommerce. Everything under the internal namespace for WooCommerce is according to their [docs](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/src/Internal#readme) subject to breaking changes without deprecation notices.

They have a utility method for this though in the [LoggingUtil](https://github.com/woocommerce/woocommerce/blame/537650180b902897980b67387943198f788d36b1/plugins/woocommerce/src/Utilities/LoggingUtil.php#L45) class that is safer to use.